C error: init fixed array of structs

### Describe the bug

 V compiler bug, C error when compiling

### Reproduction Steps
```
struct Structure1 {
	world [6]Structure2
}

struct Structure2 {
	liste []Structure3
}

struct Structure3 {
	coo []int
}

fn main() {
	app := Structure1{
		world: [6]Structure2{init: Structure2{
			liste: []Structure3{len: 1, init: Structure3{}}
		}}
	}
}
```
### Expected Behavior

I don't know since it is a V compiler bug

### Current Behavior

```
C:/TEMP/v_0/rpdc.01JS0GR37W6GNXPYNRV9BEXHK7.tmp.c:528: warning: WINVER redefined
C:/TEMP/v_0/rpdc.01JS0GR37W6GNXPYNRV9BEXHK7.tmp.c:3830: warning: implicit declaration of function 'tcc_backtrace'
C:/TEMP/v_0/rpdc.01JS0GR37W6GNXPYNRV9BEXHK7.tmp.c:5896: warning: assignment makes pointer from integer without a cast
C:/TEMP/v_0/rpdc.01JS0GR37W6GNXPYNRV9BEXHK7.tmp.c:5896: error: ',' expected (got ";")
builder error:
==================
C error found. It should never happen, when compiling pure V code.
```

### Possible Solution

_No response_

### Additional Information/Context

I used v reduce + some manual changes to give a shorter example

### V version

V 0.4.10 81fd0db.ccb3d5c

### Environment details (OS name and version, etc.)

windows, Microsoft Windows 10 Pro 19045 64-bit
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -180,12 +180,19 @@ fn (mut g Gen) fixed_array_init(node ast.ArrayInit, array_type Type, var_name st
 				g.add_commas_and_prevent_long_lines(i, info.size)
 			}
 		} else {
-			before_expr_pos := g.out.len
-			{
-				g.expr_with_init(node)
+			if g.table.final_sym(info.elem_type).kind in [.struct, .interface] {
+				for i in 0 .. info.size {
+					g.expr_with_init(node)
+					g.add_commas_and_prevent_long_lines(i, info.size)
+				}
+			} else {
+				before_expr_pos := g.out.len
+				{
+					g.expr_with_init(node)
+				}
+				sexpr := g.out.cut_to(before_expr_pos)
+				g.write_c99_elements_for_array(info.size, sexpr)
 			}
-			sexpr := g.out.cut_to(before_expr_pos)
-			g.write_c99_elements_for_array(info.size, sexpr)
 		}
 	} else if is_amp {
 		g.write('0')

--- a/vlib/v/tests/fixed_array_init_with_init_test.v
+++ b/vlib/v/tests/fixed_array_init_with_init_test.v
@@ -1,0 +1,20 @@
+struct Structure1 {
+	world [2]Structure2
+}
+
+struct Structure2 {
+	liste []Structure3
+}
+
+struct Structure3 {
+	coo []int
+}
+
+fn test_main() {
+	app := Structure1{
+		world: [2]Structure2{init: Structure2{
+			liste: []Structure3{len: 1, init: Structure3{}}
+		}}
+	}
+	assert app.world.len == 2
+}

